src/mapping-v3-core.ts

```diff
@@ -892,6 +919,225 @@ function _handleOwnershipTransferred<T>(
+      event.params._projectId,
+      event.params._index
+    );
+    assetEntity.bytecodeAddress =
+      projextExternalAssetDependency.value.bytecodeAddress;
+    assetEntity.data = projextExternalAssetDependency.value.data;
+  }
+
+  assetEntity.save();
+
+  project.externalAssetDependencyCount = BigInt.fromI32(
+    event.params._externalAssetDependencyCount
+  );
+  project.updatedAt = event.block.timestamp;
+  project.save();
+}
+
+export function handleExternalAssetDependencyRemoved(
+  event: ExternalAssetDependencyRemoved
+): void {
+  const engineFlexContract = getV3EngineFlexContract(event.address);
+  if (engineFlexContract) {
+    _handleExternalAssetDependencyRemoved(engineFlexContract, event);
+    return;
+  }
+  log.warning("[WARN] Unknown V3 coreType for contract at address {}.", [
+    event.address.toHexString()
+  ]);
+}
+
+/**
+ * Based on the way external asset dependency removal is implement on the contract
+ * we can always assume that the last index is the one being removed.
+ */
+function _handleExternalAssetDependencyRemoved<T>(
+  contract: T,
+  event: ExternalAssetDependencyRemoved
+): void {
+  if (!(contract instanceof GenArt721CoreV3_Engine_Flex)) {
+    return;
+  }
+
+  let project = Project.load(
+    generateContractSpecificId(event.address, event.params._projectId)
+  );
+
+  if (!project) {
+    log.warning(
+      "Project not found for ExternalAssetDependencyRemoved event",
+      []
+    );
+    return;
+  }
+
+  // Remove last asset entity on the project
+  const lastEntityIndex = project.externalAssetDependencyCount.minus(
+    BigInt.fromI32(1)
+  );
+  const entity = ProjectExternalAssetDependency.load(
+    generateProjectExternalAssetDependencyId(
+      project.id,
+      lastEntityIndex.toString()
+    )
+  );
+
+  if (entity) {
+    store.remove("ProjectExternalAssetDependency", entity.id);
+  }
+
+  // Refresh project external asset dependencies
+  for (
+    let index = BigInt.fromI32(0);
+    index < lastEntityIndex;
+    index = index.plus(BigInt.fromI32(1))
+  ) {
+    const assetEntity = new ProjectExternalAssetDependency(
+      generateProjectExternalAssetDependencyId(project.id, index.toString())
+    );
+
+    const contractExternalAsset = contract.try_projectExternalAssetDependencyByIndex(
+      project.projectId,
+      index
+    );
+
+    assetEntity.cid = contractExternalAsset.value.cid;
+    assetEntity.project = project.id;
+    assetEntity.index = index;
+    assetEntity.dependencyType =
+      FLEX_CONTRACT_EXTERNAL_ASSET_DEP_TYPES[
+        contractExternalAsset.value.dependencyType
+      ];
+    if (assetEntity.dependencyType === "ONCHAIN") {
+      assetEntity.bytecodeAddress = contractExternalAsset.value.bytecodeAddress;
+      assetEntity.data = contractExternalAsset.value.data;
+    }
+    assetEntity.save();
+  }
+
+  project.externalAssetDependencyCount = lastEntityIndex;
```

Comment:
nit - maybe a dev comment here explaining the last index is the previous count minus 1?